integration/query_evaluation_test.go

```diff
@@ -30,375 +28,6 @@ import (
-			err: &mongo.CommandError{
-				Code:    2,
-				Name:    "BadValue",
-				Message: `malformed mod, divisor value is invalid :: caused by :: Out of bounds coercing to integral value`,
-			},
-		},
-		"MaxInt64_floatRemainder": {
-			filter: bson.D{{"v", bson.D{{"$mod", bson.A{1, float64(math.MaxInt64)}}}}},
-			err: &mongo.CommandError{
-				Code:    2,
-				Name:    "BadValue",
-				Message: `malformed mod, remainder value is invalid :: caused by :: Out of bounds coercing to integral value`,
-			},
-		},
-		"MaxInt64_plus": {
-			filter: bson.D{{"v", bson.D{{"$mod", bson.A{9.223372036854775808e+18, 0}}}}},
-			err: &mongo.CommandError{
-				Code:    2,
-				Name:    "BadValue",
-				Message: `malformed mod, divisor value is invalid :: caused by :: Out of bounds coercing to integral value`,
-			},
-		},
-		"MaxInt64_1": {
-			filter:      bson.D{{"v", bson.D{{"$mod", bson.A{922337203685477580, 7}}}}},
-			expectedIDs: []any{"MaxInt64"},
-		},
-		"MaxInt64_2": {
-			filter:      bson.D{{"v", bson.D{{"$mod", bson.A{9.223372036854775807e+17, 7}}}}},
-			expectedIDs: []any{},
-		},
-		"MaxInt64_3": {
-			filter:      bson.D{{"v", bson.D{{"$mod", bson.A{9.223372036854775800e+17, 7}}}}},
-			expectedIDs: []any{},
-		},
-		"MaxInt64_4": {
-			filter:      bson.D{{"v", bson.D{{"$mod", bson.A{922337203, 6854775807}}}}},
-			expectedIDs: []any{},
-		},
-		"MaxInt64_overflowVerge": {
-			filter: bson.D{{"v", bson.D{{"$mod", bson.A{9.223372036854776832e+18, 0}}}}},
-			err: &mongo.CommandError{
-				Code:    2,
-				Name:    "BadValue",
-				Message: "malformed mod, divisor value is invalid :: caused by :: Out of bounds coercing to integral value",
-			},
-		},
-		"MaxInt64_overflowDivisor": {
-			filter: bson.D{{"v", bson.D{{"$mod", bson.A{9.223372036854776833e+18, 0}}}}},
-			err: &mongo.CommandError{
-				Code:    2,
-				Name:    "BadValue",
-				Message: `malformed mod, divisor value is invalid :: caused by :: Out of bounds coercing to integral value`,
-			},
-		},
-		"MaxInt64_overflowBoth": {
-			filter: bson.D{{"v", bson.D{{"$mod", bson.A{9.223372036854776833e+18, 9.223372036854776833e+18}}}}},
-			err: &mongo.CommandError{
-				Code:    2,
-				Name:    "BadValue",
-				Message: `malformed mod, divisor value is invalid :: caused by :: Out of bounds coercing to integral value`,
-			},
-		},
-		"MinInt64_Divisor": {
-			filter:      bson.D{{"v", bson.D{{"$mod", bson.A{math.MinInt64, 0}}}}},
-			expectedIDs: []any{"MinInt64", "MinInt64_float", "MinInt64_minus", "MinInt64_overflowVerge", "SmallestNonzeroFloat64", "Zero"},
-		},
-		"MinInt64_Remainder": {
-			filter:      bson.D{{"v", bson.D{{"$mod", bson.A{1, math.MinInt64}}}}},
-			expectedIDs: []any{},
-		},
-		"MinInt64_floatDivisor": {
-			filter:      bson.D{{"v", bson.D{{"$mod", bson.A{float64(math.MinInt64), 0}}}}},
-			expectedIDs: []any{"MinInt64", "MinInt64_float", "MinInt64_minus", "MinInt64_overflowVerge", "SmallestNonzeroFloat64", "Zero"},
```

Comment:
We didn't move every value to scalars.go from here. For example we don't have any `MaxInt` value with overflow and overflowVerge. If I understand they are not relevant because we stopped supporting values smaller than `MinInt64` and greater than `MaxInt64`, right?

Reply:
They are moved, but maybe you are confused with the naming. `MaxInt64_overflowVerge` and `MaxInt64_overflow` maybe called _MaxInt64..._ but they were actually float64. 
```
bson.D{{"_id", "MaxInt64_overflowVerge"}, {"v", 9.223372036854776832e+18}},
bson.D{{"_id", "MaxInt64_overflow"}, {"v", 9.223372036854776833e+18}},
```
I added them to Scalars, with _double_ prefix.
```
		"double-max-overflow":       9.223372036854776833e+18,
		"double-max-overflow-verge": 9.223372036854776832e+18,
```
They are the same number, we definitely want to support them :hugs: And adding them to scalars discovered bug related to bitwise operation which was fixed in this PR too :muscle:

Reply:
> If that's right then:

No, mod supports `MaxInt64_overflow`. The fix was for not $mod operator but for bitwise operator which started failing when I added overflow numbers to Scalars. Because I suppose there was edge case that was not covered when bitwise operator were implemented. :hugs: 

> does MongoDB support them?

Yes, and we continue to support them

> couldn't we include in scalars overflows of max/min int32 or that doesn't make any difference?

It's already there. https://github.com/FerretDB/FerretDB/blob/main/integration/shareddata/scalars.go#L77